web/src/fetch_status.ts

```diff
@@ -1,6 +1,14 @@
 import * as message_feed_loading from "./message_feed_loading";
+import type {RawMessage} from "./types";
 
-function max_id_for_messages(messages) {
+type FetchStatusOptions = {
+    found_oldest?: boolean;
+    found_newest?: boolean;
+    history_limited?: boolean;
+    update_loading_indicator: boolean;
+};
```

Comment:
I don't think type makes sense for the `start_` methods, which seem to only accept the `update_loading_indicator` parameter. I would rather we inline these for the 4 methods with the actual parameters that each function accepts.